pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -384,6 +386,155 @@ func convertGRPCRoute(r k8s.GRPCRouteRule, ctx configContext,
+	}
+
+	return drs
+}
+
+func buildBackendLBPolicyDestinationRule(r configContext, obj config.Config) []config.Config {
+	backendLBPolicy, _ := obj.Spec.(*k8salpha.BackendLBPolicySpec)
+	if backendLBPolicy.SessionPersistence == nil {
+		return nil
+	}
+	wrappedStatus, _ := obj.Status.(*kstatus.WrappedStatus)
+	if wrappedStatus.Status == nil {
+		wrappedStatus.Status = &k8salpha.PolicyStatus{}
+	}
+	reportStatus := func(results []RouteParentResult) {
+		obj.Status.(*kstatus.WrappedStatus).Mutate(func(s config.Status) config.Status {
+			rs := s.(*k8salpha.PolicyStatus)
+			rs.Ancestors = createPolicyStatus(results, obj, rs.Ancestors)
+			return rs
+		})
+	}
+
+	var err error
+	var drs []config.Config
+	var ancestorRefs []routeParentReference
+	for _, ref := range backendLBPolicy.TargetRefs {
+		dr := &istio.DestinationRule{
+			Host: fmt.Sprintf("%s.%s.svc.%s", ref.Name, obj.Namespace, r.Domain),
+		}
+		dr.TrafficPolicy = &istio.TrafficPolicy{
+			LoadBalancer: &istio.LoadBalancerSettings{},
+		}
+
+		switch *backendLBPolicy.SessionPersistence.Type {
+		case k8s.CookieBasedSessionPersistence:
+			dr.TrafficPolicy.LoadBalancer.LbPolicy, err = buildCookieBasedConsistentHash(backendLBPolicy)
+
+		case k8s.HeaderBasedSessionPersistence:
+			dr.TrafficPolicy.LoadBalancer.LbPolicy, err = buildHeaderBasedConsistentHash(backendLBPolicy)
+		}
+
+		var deniedReason *ParentError
+		if err != nil {
+			deniedReason = &ParentError{Reason: AncestorErrorInvalid, Message: err.Error()}
+		}
+		namespace := k8s.Namespace(obj.Namespace)
+		parentRef := k8s.ParentReference{
+			Group:     &ref.Group,
+			Kind:      &ref.Kind,
+			Name:      ref.Name,
+			Namespace: &namespace,
+		}
+		ir, _ := toInternalParentReference(parentRef, obj.Namespace)
+		ancestorRefs = append(ancestorRefs, routeParentReference{
+			InternalName:      string(ref.Name),
+			InternalKind:      ir.Kind,
+			Hostname:          dr.Host,
+			DeniedReason:      deniedReason,
+			OriginalReference: parentRef,
+		})
+		if deniedReason != nil {
+			continue
+		}
+		drs = append(drs, config.Config{
+			Meta: config.Meta{
+				CreationTimestamp: obj.CreationTimestamp,
+				GroupVersionKind:  gvk.DestinationRule,
+				Name:              obj.Name,
+				Namespace:         obj.Namespace,
+				Domain:            r.Domain,
+			},
+			Spec: dr,
+		})
+	}
+	reportStatus(slices.Map(ancestorRefs, func(r routeParentReference) RouteParentResult {
+		res := RouteParentResult{
+			OriginalReference: r.OriginalReference,
+			DeniedReason:      r.DeniedReason,
+		}
+		return res
+	}))
+	return drs
+}
+
+func buildCookieBasedConsistentHash(backendLBPolicy *k8salpha.BackendLBPolicySpec) (*istio.LoadBalancerSettings_ConsistentHash, error) {
+	cookieName := "http-cookie"
+	if backendLBPolicy.SessionPersistence.SessionName != nil {
+		cookieName = *backendLBPolicy.SessionPersistence.SessionName
+	}
+
+	cookieTTL, err := determineCookieTTL(backendLBPolicy)
+
+	return &istio.LoadBalancerSettings_ConsistentHash{
```

Comment:
@costinm I thought this was supposed to translate to `envoy.http.stateful_session.cookie` -- is that right or have I not kept up with the iterations on the GEP